comp/core/autodiscovery/listeners/kubelet.go

```diff
@@ -133,6 +135,11 @@ func (l *KubeletListener) createContainerService(
 		}
 	}
 
+	if !container.LastSeenReady.IsZero() && time.Since(container.LastSeenReady) > unreadinessTimeout {
```

Comment:
it would be nice to add a comment that explains why we need this specific logic here.